Classes/Formatter/DateTimeFormatter.php

```diff
@@ -27,13 +27,17 @@ class DateTimeFormatter
      * @param string|null $locale
      * The locale e.G. "en_US" to use. Will default to the current php locale.
      *
+     * @param string|null $pattern
+     * The pattern allows a custom format. @link http://php.net/manual/en/intl.intldateformatter-constants.php
+     *
      * @return string The formatted string or, if an error occurred, null
      */
     public function format(
         $value,
         int $dateType = IntlDateFormatter::MEDIUM,
         int $timeType = IntlDateFormatter::MEDIUM,
-        ?string $locale = null
+        ?string $locale = null,
+        ?string $pattern = null
```

Comment:
The combination of `$pattern` with `$dateType` and `$timeType` does not seem to make much sense. AFAIK the pattern would override `$dateType` and `$timeType`. Therefore I suggest to remove the new `$pattern` parameter here. People who would like to use the pattern, can use the new `formatWithPattern` method below.

Reply:
My idea is now to make the function `format` as `@internal` and adding an new function `formatDateTime($value, int $dateType = IntlDateFormatter::MEDIUM, int $timeType = IntlDateFormatter::MEDIUM, $locale)` as replacement for the current `format`-Function without the `$pattern`.

The function `formatWithPattern` use the internal function `format` to use correct `$locale`, so it is therefore not possible to do without the `$pattern` variable.

Reply:
If possible, I would like to keep the `format` interface, as it is. It is a public interface, so I would like to keep it stable. See !17 for an alternative. Would that suffice your needs?

Reply:
#17 looks great, thanks for your work. I close this MR and approve #17